include/pybind11/pybind11.h

```diff
@@ -887,19 +887,28 @@ class cpp_function : public function {
     }
 };
 
-/// Wrapper for Python extension modules
+/// Wrapper for Python modules
 class module_ : public object {
 public:
     PYBIND11_OBJECT_DEFAULT(module_, object, PyModule_Check)
 
     /// Create a new top-level Python module with the given name and docstring
-    PYBIND11_DEPRECATED("Use PYBIND11_MODULE or module_::create_extension_module instead")
```

Comment:
I think it's important to add a long comment here, to explain that v2.7 changes what `module_::module` does. It also needs a long-ish section in the changelog.

Reply:
Better than the changelog, even: I'll add something to the upgrade guide as well, mentioning the error you get when you would return this as extension module.

Reply:
Done!

Let me quickly still add a changelog entry. Apparently this PR is so old that it didn't even ask me on creating this PR!